app/src/features/sources/sourceSlice.ts

```diff
@@ -0,0 +1,23 @@
+import { createSlice, createAction } from "@reduxjs/toolkit";
+
+import type { RootState } from "../../app/store";
+import type { Source } from "../../services/api/generated/api.generated";
+
+const initialState = null as null | Source;
+
+export const editSource = createAction<null | Source>("editSource");
+
+const sourceSlice = createSlice({
+  name: "source",
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(editSource, (state, action) => {
+      return action.payload;
+    });
+  },
+});
```

Comment:
Is there a reason editSource action is defined outside of sourceSlice scope ?

We could directly define it as a `reducers` attribute

```suggestion
export const editSource = createAction<null | Source>("editSource");

const sourceSlice = createSlice({
  name: "source",
  initialState,
  reducers: {
     editSource: (state, action: PayloadAction<null | Source>) => {
        return action.payload;
     }
  }
});
```

Reply:
it seems to be the recommend api with Typescript. 
https://redux-toolkit.js.org/api/createSlice#the-extrareducers-builder-callback-notation
```
We recommend using this API as it has better TypeScript support (and thus, IDE autocomplete even for JavaScript users), as it will correctly infer the action type in the reducer based on the provided action creator. It's particularly useful for working with actions produced by createAction and createAsyncThunk.
```